cpp/src/arrow/adapters/orc/util.cc

```diff
@@ -951,6 +952,15 @@ Status WriteBatch(const Array& array, int64_t orc_offset,
   }
 }
 
+void SetAttributes(const std::shared_ptr<arrow::Field>& field, liborc::Type* type) {
+  if (field->HasMetadata()) {
+    const auto& metadata = field->metadata();
+    for (int j = 0; j < metadata->size(); j++) {
```

Comment:
```suggestion
    for (int64_t j = 0; j < metadata->size(); j++) {
```

Reply:
Oops, that was refactored out of an inner loop but I forgot to rename it.